docs/EventuallyConsistentServices.md

```diff
@@ -4,21 +4,42 @@ A message-based architecture (where messages are pushed onto queues and pulled o
 
 GherkinSpec includes basic support for modern message-based microservices in order to keep the "retry" complexity out of the test code.
 
-Further features are planned (see the Issues list).
+Eventual consistency is supported at two levels, step and scenario.
+
+The intended usage of step level eventual consistency is as follows:
+
+* A service consumes an event from another service, performs some processing, then publishes a different event.
+* A test exists for this behaviour: "... When event X is received, Then event Y is published."
+* Step level eventual consistency may be applied on the Then step in this test to cause the Then step to be retried until the event has been published.
+
+The intended usage of scenario level eventual consistency is as follows:
+
+* A service accepts data via HTTP PUT. The service enqueues a command to process that data to immediately return to the caller.  The service then serves the processed data via HTTP GET.
+* A test exists for this behaviour: "Given I have submitted data, When I retrieve that data, Then the data matches my expectation."
+* Scenario level eventual consistency may be applied to this test scenario to cause the When and Then steps to be retried until the data has been processed and is being served via HTTP GET.
+
+Both levels of eventual consistency may be combined, as described in the following example:
```

Comment:
@DaveAurionix do you think this is too confusing? If so we could remove this and the related gotcha below

Reply:
As discussed, we'll remove this.

Reply:
Removed as discussed